### PR TITLE
Add Ocl2Hcl.

### LINKED ISSUE #23: Translate OCL to Plugin description for Huawei Cloud
## Task List

* [ ] Translate image block to Packer description.
* [ ] Translate vm block to Terraform description.
* [ ] Translate network block to Terraform description.
* [ ] Translate storage block to Terraform description.

### PR DESCRIPTION
**Pull request by iskey**
Add OCL2HCL translations.
fix #23

### CODE REVIEW

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/builders/terraform/Ocl2Hcl.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+        return portPairs;
+    }
+
+    public String getHclSecurityGroupRule() {
+        if (ocl == null
+            || ocl.getNetwork() == null
+            || ocl.getNetwork().getSecurity() == null
+            || ocl.getNetwork().getSecurity().get(0).getRules() == null) {
+            return "Invalid Security Group Rule";
```
Comment (Jiajia-Wen) [inline services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/builders/terraform/Ocl2Hcl.java:62]: An exception is better than return error code in Java.

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/builders/terraform/Ocl2Hcl.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            hcl += getHclVpc();
+            hcl += getHclVm();
+            hcl += getHclVpcSubnet();
+            hcl += getHclFlavor();
+            hcl += getHclImage();
+            hcl += getHclAvailabilityZone();
+            hcl += getHclStorage();
+            return hcl;
+        } catch (Exception ex) {
+            log.error("Ocl2Hcl failed. Please check you OCL.", ex);
```
Comment (Jiajia-Wen) [inline services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/builders/terraform/Ocl2Hcl.java:250]: throw an exception here.

**Review by Jiajia-Wen — COMMENTED** [review]
General comments:  in Java,  when error happens,  normally an exception should be thrown,  t,

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/AtomBuilder.java [hunk 3]
```diff
@@ -40,48 +40,73 @@ public AtomBuilder(Ocl ocl) {
...
-                    .getTimeout();
+            timeToWait = subBuilders.stream()
+                .max(Comparator.comparing(AtomBuilder::getTimeout))
+                .get()
+                .getTimeout();
         }
 
         while (!subBuilders.isEmpty() && subBuilders.stream().allMatch(AtomBuilder::needWaiting)) {
             try {
                 TimeUnit.SECONDS.sleep(5);
```
[the remark this answers is not in the document]
Comment (iskey) [reply services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/AtomBuilder.java]: Good idea.

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java [hunk 4]
```diff
@@ -22,55 +25,76 @@ public String name() {
...
+
+        config = configService;
     }
 
     @Override
     public void registerManagedService(Ocl ocl) {
-        managedOcl.put(ocl.getName(), ocl);
         log.info("Register managed service, creating  Huawei Cloud resource");
+        if (ocl == null) {
+            log.error("registering invalid ocl. ocl = null");
```
[the remark this answers is not in the document]
Comment (iskey) [reply services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java]: Yup!

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java [hunk 5]
```diff
@@ -22,55 +25,76 @@ public String name() {
...
     public void startManagedService(String managedServiceName) {
+        log.info("Start managed service {} on Huawei Cloud", managedServiceName);
         if (!managedOcl.containsKey(managedServiceName)) {
-            log.warn("Service: {} not registered.", managedServiceName);
+            log.error("Service: {} not registered.", managedServiceName);
             return;
         }
+
+        // TODO: need to do a deep copy for ocl.
+        Ocl ocl = managedOcl.get(managedServiceName);
```
[the remark this answers is not in the document]
Comment (iskey) [reply services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java]: Ok, I will refine it.

File: services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java [hunk 6]
```diff
@@ -22,55 +25,76 @@ public String name() {
...
+        if (ocl == null) {
+            log.error("Ocl object is null.");
+            return;
+        }
+
         BuilderFactory factory = new BuilderFactory();
-        Optional<AtomBuilder> optionalAtomBuilder = factory.createBuilder("basic", managedOcl.get(managedServiceName));
+        Optional<AtomBuilder> optionalAtomBuilder = factory.createBuilder("basic", ocl);
```
[the remark this answers is not in the document]
Comment (iskey) [reply services/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java]: ok

**Review by Jiajia-Wen — APPROVED** [review]
General comment: can add more "input" in the illegalArgumentException to indicate what is illegal

**Pull request merged**